### ISSUE TITLE
[a11y]: DataTable [with-selection-sorting] - sorting controls in the column header 

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#14273]
### Package

@carbon/react

### Browser

FF

### Package version

@carbon/react@1.32.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description
Sorting controls in the column header

It would be better if the label remains static because the toggle status is already conveyed by the "check" / "un checked" announcement.

JAWS announces the entire column content when we activate the sorting controls each time. However this issue is not consistent. Note: Same issue is found in the DataTable / Sorting - Default  as well

### Recommended Fix
remove the extraneous announcement

Any redundant, excessively wordy and extraneous announcement can be distracting for the screen reader users. 

### WCAG 2.1 Violation
4.2.1 Name, Role, and Value

### Reproduction/example
Record `Number 78` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

https://react.carbondesignsystem.com/?path=/story/components-datatable-selection--with-selection-and-sorting

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by heloiselui** [comment]
Checked with our accessibility team, and this issue still exists.
